### PR TITLE
types: add FastifyJwtNamespace helper

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -472,6 +472,18 @@ fastify.post('/sign/:namespace', async function (request, reply) {
 })
 ```
 
+#### Typescript
+
+To simplify the usage of namespaces in typescript, you can use the `FastifyJwtNamespace` helper type:
```
Comment (Fdawgs) [inline README.md:477]:
```suggestion
To simplify the use of namespaces in TypeScript you can use the `FastifyJwtNamespace` helper type:
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Comment by Uzlopak** [comment]
@Fdawgs 

Thanks

**Review by Fdawgs — APPROVED** [review]
LGTM on readme.

File: types/jwt.test-d.ts [hunk 2]
```diff
@@ -10,19 +10,19 @@ const secretOptions = {
     public: 'publicKey',
     private: 'privateKey'
   },
-  secretFnCallback: (_req, _token, cb) => { cb(null, 'supersecret') },
-  secretFnPromise: (_req, _token) => Promise.resolve('supersecret'),
-  secretFnAsync: async (_req, _token) => 'supersecret',
-  secretFnBufferCallback: (_req, _token, cb) => { cb(null, Buffer.from('some secret', 'base64')) },
-  secretFnBufferPromise: (_req, _token) => Promise.resolve(Buffer.from('some secret', 'base64')),
-  secretFnBufferAsync: async (_req, _token) => Buffer.from('some secret', 'base64'),
+  secretFnCallback: (_req: any, _token: any, cb: any) => { cb(null, 'supersecret') },
```
Comment (climba03003) [inline types/jwt.test-d.ts:13]: Shouldn't the function arguments should infer the correct types here?
Providing `:any` seems to be breaking changes.

We may type the `const` as `Record<string, FastifyJWTOptions['secret']`

Comment (Uzlopak) [reply types/jwt.test-d.ts:13]: Makes sense

**Comment by Uzlopak** [comment]
@climba03003 

I somehow think that all plugins which use namespaces or custom defined values should have something like this helper interface.

I am currently thinking, if instead of having four generics, to have one generic and use that for the interface. Interface would be then identical to the option.

File: types/jwt.test-d.ts (hunk 2 above)
Comment (Uzlopak) [reply types/jwt.test-d.ts:13]: We have again issues with determining the right function type

**Pull request merged**